packages/server/src/stream/service.ts

```diff
@@ -0,0 +1,132 @@
+import { HealthEvents } from '../health/events'
+import { HealthService } from '../health/service'
+import { Logger } from '../logger/types'
+import { MappingService } from '../mapping/service'
+import { MessageEvents } from '../messages/events'
+import { MessageService } from '../messages/service'
+import { PostService } from '../post/service'
+import { SocketService } from '../socket/service'
+import { UserEvents } from '../users/events'
+import { UserService } from '../users/service'
+import { WebhookService } from '../webhooks/service'
+
+export class StreamService extends Service {
+  private logger = new Logger('Stream')
+
+  constructor(
+    private posts: PostService,
+    private sockets: SocketService,
+    private channels: ChannelService,
+    private clients: ClientService,
+    private webhooks: WebhookService,
+    private conduits: ConduitService,
+    private health: HealthService,
+    private users: UserService,
+    private conversations: ConversationService,
+    private messages: MessageService,
+    private mapping: MappingService
+  ) {
+    super()
+  }
+
+  async setup() {
+    this.health.events.on(HealthEvents.Registered, async ({ event }) => {
+      const conduit = await this.conduits.get(event.conduitId)
+      const client = await this.clients.getByProviderId(conduit!.providerId)
+      if (!client) {
+        return
+      }
+
+      const channel = this.channels.getById(conduit!.channelId)
+      await this.stream(
+        'health.new',
+        { channel: channel.name, event: { ...this.health.makeReadable(event) } },
+        client.id,
+        undefined
+      )
+    })
+
+    this.users.events.on(UserEvents.Created, async ({ user }) => {
+      await this.stream('user.new', {}, user.clientId, user.id)
+    })
+
+    this.conversations.events.on(ConversationEvents.Created, async ({ conversation }) => {
+      await this.stream(
+        'conversation.new',
+        { conversationId: conversation.id },
+        conversation.clientId,
+        conversation.userId
+      )
+    })
+
+    this.messages.events.on(MessageEvents.Created, async ({ message, source }) => {
+      const conversation = await this.conversations.get(message.conversationId)
+
+      await this.stream(
+        'message.new',
+        { channel: await this.getChannel(conversation!.id), conversationId: conversation!.id, message },
+        conversation!.clientId,
+        conversation!.userId,
+        source
+      )
+    })
+  }
+
+  private async getChannel(conversationId: uuid) {
+    const convmaps = await this.mapping.convmap.listByConversationId(conversationId)
+    if (convmaps.length === 1) {
+      const tunnel = await this.mapping.tunnels.get(convmaps[0].tunnelId)
+      return this.channels.getById(tunnel!.channelId).name
+    } else {
+      return 'messaging'
+    }
+  }
+
+  async stream(type: string, payload: any, clientId: uuid, userId: uuid | undefined, source?: ActionSource) {
```

Comment:
```suggestion
  async stream(type: string, payload: any, clientId: uuid, userId?: uuid, source?: ActionSource) {
```

You can then remove the undefined on line 55